iowrappers/nearby_search.go

```diff
@@ -32,19 +34,37 @@ type PlaceSearchRequest struct {
 	BusinessStatus POI.BusinessStatus
 	// true if using precise geolocation instead of using a grander administrative area
 	UsePreciseLocation bool
+
+	PriceLevel POI.PriceLevel
 }
 
-func (c *MapsClient) GoogleMapsNearbySearchWrapper(ctx context.Context, location POI.Location, placeType string, radius uint, pageToken string) (resp maps.PlacesSearchResponse, err error) {
-	mapsReq := maps.NearbySearchRequest{
-		Type: maps.PlaceType(placeType),
+// Create NearbySearchRequest for maps NearbySearch, adjust key settings such as radius and price levels
+func CreateMapSearchRequest(reqIn *PlaceSearchRequest, placeType POI.LocationType, token string) (reqOut maps.NearbySearchRequest) {
+	// Adjust radius, minPrice and maxPrice settings in search request
+	var radius uint = reqIn.Radius
+	var minPrice, maxPrice maps.PriceLevel
+	if POI.IsPricyEatery(reqIn.PlaceCat, reqIn.PriceLevel) {
+		// increase search radius
+		radius = reqIn.Radius * 4
```

Comment:
We want to make sure the radius does not exceed the maximum [supported](https://developers.google.com/maps/documentation/places/web-service/search-nearby#radius) by Google Maps.

Reply:
Done, added a min function to set a limit with GoogleMap max radius [7ba29b3](https://github.com/timwangmusic/Vacation-Planner/pull/282/commits/7ba29b3986970f85256a8900849d1d7433eb8274)